### PR TITLE
[DROOLS-3487] Add support to list and map to scesim runner

### PR DESCRIPTION
**Pull request by danielezonca**
This PR contains runner support for List/Map in both Rule and DMN scenarios

https://issues.jboss.org/browse/DROOLS-3487

To be merged after https://github.com/kiegroup/drools-wb/pull/1045

@kkufova @gitgabrio @Rikkola

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenterTest.java [hunk 1]
```diff
@@ -194,7 +191,7 @@ protected void clearRightPanelStatus() {
 
 //            @Override
 //            protected void loadContent() {
-//                return new ScenarioSimulationModelContent(model,
+//                return new ScenarioSimulationModelContent(modelLocal,
```
Comment (Rikkola) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenterTest.java]: Some old commented code that could be removed.

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/ItemElementViewImplTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public void setup() {
+        elementPresenterMock = mock(ItemElementView.Presenter.class);
+        elementView = spy(new ItemElementViewImpl() {
+            {
+                this.presenter = elementPresenterMock;
+                this.faAngleRight = faAngleRightMock;
+            }
+        });
+    }
+
```
[the remark this answers is not in the document]
Comment (gitgabrio) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/ItemElementViewImplTest.java]: @Rikkola 
Those two classes are actual instances of an (abstract) parent.
They share same methods but on different objects, so logic is
1) set the correct object under test in concrete classes
2) implement test only once in parent (abstract) class

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/KeyValueElementViewImplTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        elementPresenterMock = mock(KeyValueElementView.Presenter.class);
+        elementView = spy(new KeyValueElementViewImpl() {
+            {
+                this.presenter = elementPresenterMock;
+                this.faAngleRight = faAngleRightMock;
+            }
+        });
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (gitgabrio) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/KeyValueElementViewImplTest.java]: @Rikkola 
See above

**Comment by danielezonca** [comment]
Merged into https://github.com/kiegroup/drools-wb/pull/1057

**Pull request closed without merging**